lib/auth/keystore/gcp_kms_test.go

```diff
@@ -593,3 +589,155 @@ func TestGCPKMSKeystore(t *testing.T) {
 		})
 	}
 }
+
+func TestGCPKMSDeleteUnusedKeys(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	localHostID := "local-host-id"
+	otherHostID := "other-host-id"
+	localKeyring := "local-keyring"
+	otherKeyring := "other-keyring"
+
+	for _, tc := range []struct {
+		desc            string
+		existingKeys    []keySpec
+		activeKeys      []keySpec
+		expectDestroyed []keySpec
+	}{
+		{
+			// Only inactive keys should be destroyed.
+			desc: "active and inactive",
+			existingKeys: []keySpec{
+				{keyring: localKeyring, id: "id_active", host: localHostID},
+				{keyring: localKeyring, id: "id_inactive", host: localHostID},
+			},
+			activeKeys: []keySpec{
+				{keyring: localKeyring, id: "id_active", host: localHostID},
+			},
+			expectDestroyed: []keySpec{
+				{keyring: localKeyring, id: "id_inactive", host: localHostID},
+			},
+		},
+		{
+			// Inactive key from other host should not be destroyed, it may
+			// be recently created and just not added to Teleport CA yet, or the
+			// other Auth might be in a completely different Teleport cluster
+			// using the same keyring (I wouldn't advise this but someone might
+			// do it).
+			desc: "inactive key from other host",
+			existingKeys: []keySpec{
+				{keyring: localKeyring, id: "id_inactive_local", host: localHostID},
+				{keyring: localKeyring, id: "id_inactive_remote", host: otherHostID},
+			},
+			expectDestroyed: []keySpec{
+				{keyring: localKeyring, id: "id_inactive_local", host: localHostID},
+			},
+		},
+		{
+			// The presence of active keys created by a remote host in the local
+			// keyring should not break the DeleteUnusedKeys operation.
+			desc: "active key from other host",
+			existingKeys: []keySpec{
+				{keyring: localKeyring, id: "id_local_active", host: localHostID},
```

Comment:
nit: just to be consistent

```suggestion
				{keyring: localKeyring, id: "id_active_local", host: localHostID},
```